config/routes.rb

```diff
@@ -1,10 +1,14 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-  get '/users', to: 'users#index'
+  get '/users', to: 'users#index', as: 'users'
   get '/users/:id', to: 'users#show'
   get '/users/:user_id/posts', to: 'posts#index'
   get '/users/:user_id/posts/:id', to: 'posts#show'
-
+  get '/posts/new', to: 'posts#new', as: 'new_post'
+  post '/posts/create', to: 'posts#create', as: 'create_post'
+  get '/users/:user_id/posts/:id/new_comment', to: 'posts#new_comment', as: 'new_comment'
+  post '/users/:user_id/posts/:id/create_comment', to: 'posts#create_comment', as: 'create_comment'
+  post '/users/:user_id/posts/:id/like', to: 'posts#like', as: 'like_post'
```

Comment:
- [ ] We can clean up these routes based on the controller we defined, above:

  ```
  resources :users do
    resources :posts do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:create]
    end
  end
  ```

Reply:
This change has been done well